stix2validator/shoulds.py

```diff
@@ -274,14 +274,16 @@ def relationships_strict(instance):
         return
 
     r_type = instance['relationship_type']
-    r_source = re.search("(.+)\-\-", instance['source_ref']).group(1)
-    r_target = re.search("(.+)\-\-", instance['target_ref']).group(1)
+    try:
+        r_source = re.search("(.+)\-\-", instance['source_ref']).group(1)
+        r_target = re.search("(.+)\-\-", instance['target_ref']).group(1)
+    except (AttributeError, TypeError):
```

Comment:
`split('--')[0]` won't catch the problem of someone putting in a string that doesn't contain '--'